src/document/Document.js

```diff
@@ -698,7 +698,24 @@ define(function (require, exports, module) {
     Document.prototype.isUntitled = function () {
         return this.file instanceof InMemoryFile;
     };
+
+    /**
+     * Set lineEnding of the document. Change Text in the process.
+     */
+    Document.prototype.setLineEndings = function (lineEndings) {
+        this._lineEndings = lineEndings;
+        this.setText(FileUtils.translateLineEndings(this.getText(true), lineEndings));
```

Comment:
If we avoid this, the new lineEndings is not set to the existing file text. And document is left unmodified. After this, file gets dirty and once we save, the new line endings is saved to file. It doesnt happen otherwise.